test/specs/events.spec.js

```diff
@@ -1,56 +1,224 @@
-              scaleID: 'y',
-              value: 3
-            },
-            {
-              type: 'line',
-              scaleID: 'y',
-              value: 3,
-              enter: () => true
-            },
-            {
-              type: 'line',
-              scaleID: 'y',
-              value: 3
-            }
-          ]
+      }
+    };
+    it('should detect listened and moveListened state set to true', function() {
+      const chart = window.acquireChart(chartConfig);
+      const Annotation = window['chartjs-plugin-annotation'];
+      const state = Annotation._getState(chart);
+      expect(state.listened).toBe(true);
+      expect(state.moveListened).toBe(true);
+    });
+  });
+
+  describe('check', function() {
+    const chartConfig = {
+      type: 'scatter',
+      options: {
+        animation: false,
+        scales: {
+          x: {
+            display: false,
+            min: 0,
+            max: 10
+          },
+          y: {
+            display: false,
+            min: 0,
+            max: 10
+          }
+        },
+        plugins: {
+          legend: false,
+          annotation: {
+          }
         }
       }
-    },
-  };
+    };
+    const options = {
+      type: 'box',
+      id: 'test',
+      xScaleID: 'x',
+      yScaleID: 'y',
+      xMin: 2,
+      yMin: 2,
+      xMax: 4,
+      yMax: 4
+    };
+    const x0y0 = {x: 0, y: 0};
+    const center = function(chart) {
+      const xCenter = options.xMax - options.xMin;
+      const yCenter = options.yMax - options.yMin;
+      const xScale = chart.scales.x;
+      const yScale = chart.scales.y;
+      return {x: xScale.getPixelForValue(xCenter), y: yScale.getPixelForValue(yCenter)};
+    };
 
-  it('should not throw any exception', function() {
+    const pluginOpts = chartConfig.options.plugins.annotation;
 
-    const chart = window.acquireChart(chartConfig);
-    const Annotation = window['chartjs-plugin-annotation'];
-    const state = Annotation._getState(chart);
+    [pluginOpts, options].forEach(function(targetOptions) {
+      it('should not detect any move events (because no callback is set)', function(done) {
+        const enterSpy = jasmine.createSpy('enter');
+        const leaveSpy = jasmine.createSpy('leave');
 
-    expect(state.listened).toBe(true);
-    expect(state.moveListened).toBe(true);
+        pluginOpts.annotations = [options];
 
-  });
+        const chart = window.acquireChart(chartConfig);
+        window.triggerMouseEvent(chart, 'mousemove', center(chart));
+        window.afterEvent(chart, 'mousemove', function() {
+          expect(enterSpy.calls.count()).toBe(0);
+
+          window.triggerMouseEvent(chart, 'mousemove', x0y0);
+
+          window.afterEvent(chart, 'mousemove', function() {
+            expect(leaveSpy.calls.count()).toBe(0);
+            done();
+          });
+        });
+      });
+
+      it('should not detect any move events (because only click is set)', function(done) {
```

Comment:
This test seems similar, a spy that is not attached, does not really test anything (except maybe some of jasmine's code)

Reply:
This test has been added to test an event type not managed. We can remove but a part of the code is not tested (when we test the event type).

https://github.com/chartjs/chartjs-plugin-annotation/blob/d9996081e8e7bc896dccdbb0e811fa237a28b7d8/src/events.js#L46-L59

I can remove it but the coverall will provide a warning for missing tests.